File: src/main/java/mediathek/gui/actions/ShowProgramInfosAction.java
package mediathek.gui.actions;

import mediathek.update.ProgrammUpdateSuchen;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class ShowProgramInfosAction extends AbstractAction {
    public ShowProgramInfosAction() {
        putValue(Action.NAME,"Programminfos anzeigen...");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        new ProgrammUpdateSuchen().checkVersion(false, true, false, false);
    }
}

File: src/main/kotlin/mediathek/gui/actions/ShowProgramInfosAction.kt
package mediathek.gui.actions

import mediathek.update.ProgrammUpdateSuchen
import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.JFrame

class ShowProgramInfosAction(
    private val parent: JFrame,
) : AbstractAction() {
    init {
        putValue(NAME, "Programminfos anzeigen...")
    }

    override fun actionPerformed(event: ActionEvent?) {
        ProgrammUpdateSuchen(ownerProvider = { parent }).checkVersion(false, true, false, false)
    }
}
